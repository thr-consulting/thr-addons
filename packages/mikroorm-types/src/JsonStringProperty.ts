import {Type} from '@mikro-orm/core';

// NOTE this property only converts to json string, but leaves it as a string when retrieving it from the DB
// We want it to stay like this!!!
export class JsonStringProperty extends Type<any, string> {
	convertToDatabaseValue(javascriptValue: any) {
		return JSON.stringify(javascriptValue);
	}

	convertToJSValue(databaseValue: string) {
		return databaseValue;
	}

	getColumnType() {
		return 'jsonb';
	}
}
