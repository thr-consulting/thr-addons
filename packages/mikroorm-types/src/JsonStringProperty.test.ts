import {JsonStringProperty} from './JsonStringProperty';

describe('JsonStringProperty', () => {
	it('should convert values', () => {
		const obj = {
			field1: 'field1value',
			field2: {
				field3: 'field3value',
			},
		};

		const prop = new JsonStringProperty();
		expect(prop.convertToDatabaseValue(obj)).toMatchSnapshot();
		expect(prop.convertToJSValue(JSON.stringify(obj))).toMatchSnapshot();
		expect(prop.getColumnType()).toMatchSnapshot();
	});
});
