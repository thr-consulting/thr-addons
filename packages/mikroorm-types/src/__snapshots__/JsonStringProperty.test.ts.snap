// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonStringProperty should convert values 1`] = `"{\\"field1\\":\\"field1value\\",\\"field2\\":{\\"field3\\":\\"field3value\\"}}"`;

exports[`JsonStringProperty should convert values 2`] = `"{\\"field1\\":\\"field1value\\",\\"field2\\":{\\"field3\\":\\"field3value\\"}}"`;

exports[`JsonStringProperty should convert values 3`] = `"jsonb"`;
